scripts/xtalk_filter_run.py

```diff
@@ -23,39 +21,50 @@
     if a.window == 'tukey':
         filter_kwargs['alpha'] = a.alpha
     avg_red_bllens=False
-    skip_gaps_larger_then_filter_peri=False
+    skip_gaps_larger_then_filter_period=False
     skip_flagged_edges=False
     max_contiguous_edge_flags=10000
     flag_model_rms_outliers=False
 elif a.mode == 'dayenu':
     filter_kwargs = {'tol': a.tol, 'max_frate_coeffs': a.max_frate_coeffs}
     avg_red_bllens=True
     max_contiguous_edge_flags=10000
-    skip_gaps_larger_then_filter_peri=False
+    skip_gaps_larger_then_filter_period=False
     skip_flagged_edges=False
     flag_model_rms_outliers=False
 elif a.mode == 'dpss_leastsq':
     filter_kwargs = {'tol': a.tol, 'max_frate_coeffs': a.max_frate_coeffs}
     avg_red_bllens=True
-    skip_gaps_larger_then_filter_peri=True
+    skip_gaps_larger_then_filter_period=True
     skip_flagged_edges=True
     max_contiguous_edge_flags=1
     flag_model_rms_outliers=True
 
 
-# Run XTalk Filter
-xtalk_filter.load_xtalk_filter_and_write(a.infilename, calfile=a.calfile, avg_red_bllens=avg_red_bllens,
-                                         Nbls_per_load=a.partial_load_Nbls, spw_range=a.spw_range,
-                                         cache_dir=a.cache_dir, res_outfilename=a.res_outfilename,
-                                         clobber=a.clobber, write_cache=a.write_cache,
-                                         read_cache=a.read_cache, mode=a.mode,
-                                         factorize_flags=a.factorize_flags, time_thresh=a.time_thresh,
-                                         max_contiguous_edge_flags=max_contiguous_edge_flags,
-                                         add_to_history=' '.join(sys.argv), verbose=a.verbose,
-                                         skip_flagged_edges=skip_flagged_edges,
-                                         skip_contiguous_flags=skip_gaps_larger_then_filter_peri,
-                                         flag_yaml=a.flag_yaml,
-                                         external_flags=a.external_flags, skip_if_flag_within_edge_distance=a.skip_if_flag_within_edge_distance,
-                                         overwrite_flags=a.overwrite_flags,
-                                         flag_model_rms_outliers=flag_model_rms_outliers,
-                                         clean_flags_in_resid_flags=True, **filter_kwargs)
+if args.cornerturnfile is not None:
+    baseline_list = io.baselines_from_filelist_position(filename=a.cornerturnfile, filelist=a.datafilelist)
+else:
+    baseline_list = None
+
+if len(baseline_list) > 0:
```

Comment:
Will this break if `baseline_list` is `None?

Reply:
Yes. This existed to prevent length zero baseline lists to not throw errors, which can happen when you have more files in your data then baselines (common for two-integration files and autocorrs). I've moved the handling of this case into the covered code with some helpful warnings.